File: Backend/UtilityBillingSystem.Core/Entities/UtilityType.cs
namespace UtilityBillingSystem.Core.Entities
{
    public class UtilityType
    {
        public int UtilityTypeId { get; set; }
        public string Name { get; set; }  // Electricity, Water, Gas
        public string UnitOfMeasurement { get; set; }  // kWh, Liters, Cubic Meters
        public string Description { get; set; }
        public decimal BaseRate { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreatedAt { get; set; }

        // Navigation
        public ICollection<Tariff> Tariffs { get; set; }
        public ICollection<Connection> Connections { get; set; }
    }
}

File: utility-billing-system/src/app/core/models/utility-type.ts
export interface UtilityType {
  utilityTypeId: number;
  name: string;
  unitOfMeasurement: string;
  description: string;
  baseRate: number;
  isActive: boolean;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateUtilityTypeRequest {
  name: string;
  unitOfMeasurement: string;
  description: string;
  baseRate: number;
}

export interface UpdateUtilityTypeRequest {
  name: string;
  unitOfMeasurement: string;
  description: string;
  baseRate: number;
  isActive: boolean;
}